### ISSUE TITLE
Fork for JSON content > 2 GB

### ISSUE THREAD

**Issue by develancer** [doubledutch/LazyJSON#21]
Hi guys,

Thank you for this amazing library. It does work fast and surprisingly stable :-)

However, I've had issues with reading JSON files larger than 2 GB. The reason is simple—one cannot create a String object of such size. I therefore forked this project and adapted it to reading directly from files: https://github.com/develancer/LazyJSON64

There are some downsides, at the moment—for example, the structures are read-only and there is no „dirty” mode.

Do you think that integrating these changes into mainstream some day would be useful? We can leave it as a separate fork for now.

**Comment by kasperjj** [comment]
Absolutely! Super interesting fork... I had started down the same path a while ago to allow for parsing json without bringing it all into memory and to allow lazy decoding of unicode characters.... ie. we don't actually need to spend time decoding the whole data chunk, only string keys and values when needed. Let me dig into my old branch next week and see if we can make a shared path forward that accomplishes both!
